src/utilities/createRouter.ts

```diff
@@ -41,38 +41,35 @@ export function createRouter<const T extends Routes>(routes: T, options: RouterO
 
     const beforeResponse = await runBeforeRouteHooks({ to, from, hooks })
 
-    const setHistory = (): void => history.update(url, { replace })
-    const setRoute = (): void => updateRoute(to)
-
     switch (beforeResponse.status) {
       // On abort do nothing
       case 'ABORT':
         return
 
       // On push update the history, and push new route, and return
       case 'PUSH':
-        setHistory()
+        history.update(url, { replace })
         push(...beforeResponse.to)
         return
 
       // On reject update the history, the route, and set the rejection type
       case 'REJECT':
-        setHistory()
-        setRoute()
+        history.update(url, { replace })
         setRejection(beforeResponse.type)
         break
 
       // On success update history, set the route, and clear the rejection
       case 'SUCCESS':
-        setHistory()
-        setRoute()
+        history.update(url, { replace })
         setRejection(null)
         break
 
       default:
         throw new Error(`Switch is not exhaustive for before hook response status: ${JSON.stringify(beforeResponse satisfies never)}`)
     }
+  }
 
+  async function afterUpdate(to: ResolvedRoute, from: ResolvedRoute): Promise<void> {
```

Comment:
What is your thought process behind separating update and afterUpdate? Updating is a single action which includes running before and after hooks.